include/wx/gtk/textctrl.h

```diff
@@ -234,6 +236,7 @@ class WXDLLIMPEXP_CORE wxTextCtrl: public wxTextCtrlBase
     // a dummy one when frozen
     GtkTextBuffer *m_buffer;
 
+    bool m_maxLengthAllowed = false;
```

Comment:
why is this extra variable needed? Can't you just check if max length > 0? I removed all the `IsMaxLengthAllowed` and `m_maxLengthAllowed` and it still seems to work fine.

Reply:
@MaartenBent ,
All 3 major events - typing on the keyboard, pasting from the clipboard and setting/appending text programmatically goes thru this routine.

Now, while first 2 are fine - third one is not.
Try to run the test without this - it will fail.

The reason for that is because MS, for whatever reason, decided to check the max length for multiline text control only on the user action.
So in order for the test to pass I need to add the extra variable.

Just try to compare the test run on MS and GTK without this variable.